ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/LedgerDao.scala

```diff
@@ -25,16 +28,41 @@ final case class Contract(
     transactionId: String,
     workflowId: String,
     witnesses: Set[Ref.Party],
+    divulgences: Set[Ref.Party],
     coinst: ContractInst[VersionedValue[AbsoluteContractId]],
     key: Option[KeyWithMaintainers[VersionedValue[AbsoluteContractId]]]) {
   def toActiveContract: ActiveContract =
-    // TODO SC store divulgences
-    ActiveContract(let, transactionId, workflowId, coinst, witnesses, Set.empty, key)
+    ActiveContract(let, transactionId, workflowId, coinst, witnesses, divulgences, key)
 }
 
 object Contract {
   def fromActiveContract(cid: AbsoluteContractId, ac: ActiveContract): Contract =
-    Contract(cid, ac.let, ac.transactionId, ac.workflowId, ac.witnesses, ac.contract, ac.key)
+    Contract(
+      cid,
+      ac.let,
+      ac.transactionId,
+      ac.workflowId,
+      ac.witnesses,
+      ac.divulgences,
+      ac.contract,
+      ac.key)
+}
+
+/**
+  * Every time the ledger persists a transactions, the active contract set (ACS) is updated.
+  * Updating the ACS requires knowledge of blinding info, which is not included in LedgerEntry.Transaction.
+  * The SqlLedger persistence queue Transaction elements are therefore enriched with blinding info.
+  */
+sealed abstract class PersistenceEntry extends Product with Serializable
```

Comment:
@gaboraranyossy-da Persistence queue now stores PersistenceEntry instead of LedgerEntry, see comment.